packages/strapi-database/lib/utils/lifecycles.js

```diff
@@ -3,6 +3,10 @@
 const _ = require('lodash');
 
 const executeLifecycle = async (lifecycle, model, ...args) => {
+  // Run registered lifecycles
+  await strapi.db.lifecycles.run(lifecycle, model, ...args);
```

Comment:
Is there any specific reason why we execute the registered ones first, or is it arbitrary?

Reply:
I chose to put it first so the user gets the data in it's "final" intended form after passing through all the lifecycles. It should give more flexibility.